packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,294 @@
+    it("Container loaded before sweep timeout expires can't send ops for tombstoned datastores", async () => {
+        const { testDataObjectId, summaryVersion } = await createUnreferencedDataStore();
+
+        // Wait some time, the datastore should not be sweep ready after this wait
+        await delay(sweepTimeoutMs - wait);
+
+        // Load a new container and summarizer based on the latest summary, summarize
+        const {
+            container: summarizingContainer2,
+            summarizer: summarizer2,
+        } = await loadSummarizerAndContainer(summaryVersion);
+
+        // Use the request pattern to get the testDataObject - this is unsafe and no one should do this in their
+        // production application.
+        const testDataObject2 = await requestFluidObject<TestDataObject>(summarizingContainer2, testDataObjectId);
+
+        // Wait enough time so that the datastore is sweep ready
+        await delay(wait);
+
+        // Send an op to update the currentTimestampMs to now
+        const mainDataStore2 = await requestFluidObject<TestDataObject>(summarizingContainer2, "default");
+        mainDataStore2._root.set("send a", "op");
+
+        // The datastore should be tombstoned now
+        await summarize(summarizer2);
+
+        // The request pattern should fail!
+        assert.throws(() => testDataObject2._root.set("send", "op"),
+            (error) => {
+                const correctErrorType = error.errorType = "dataCorruptionError";
+                const correctErrorMessage = error.errorMessage?.startsWith(`Context was tombstoned`) === true;
+                return correctErrorType && correctErrorMessage;
+            },
+            `Should not be able to send ops for a tombstoned datastore.`);
+    });
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sessionExpiryTimeoutMs` above
+    itExpects("Container loaded after sweep timeout expires can't receive ops for tombstoned datastores",
+    [
+        {
+            eventName: "fluid:telemetry:ContainerRuntime:GarbageCollector:SweepReadyObject_Loaded",
+        },
+        {
+            eventName: "fluid:telemetry:Container:ContainerClose",
+            error: "Context was tombstoned during call of process!",
+            errorType: "dataCorruptionError",
+        },
+    ],
+    async () => {
+        const { testDataObjectId, summaryVersion } = await createUnreferencedDataStore();
+
+        // Wait a sweep worthy amount of time (all containers should have closed by now)
+        await delay(sweepTimeoutMs);
+
+        // Load a new container and summarizer based on the latest summary, summarize
+        const {
+            container: summarizingContainer2,
+            summarizer: summarizer2,
+        } = await loadSummarizerAndContainer(summaryVersion);
+
+        const container2 = await provider.loadContainer(runtimeFactory, { configProvider }, { summaryVersion });
```

Comment:
What is functionally different between `container2` and `summarizingContainer2`? They're both interactive containers loaded from the same summary, right?

Reply:
Ohh I am wrong, `summarizingContainer` is the summarizer container, and `summarizer` is the ISummarizer, not the container.  Got it.